### PR TITLE
chore: some repository cleanup

### PR DESCRIPTION
**Pull request by mlegner**
## Description

This PR contains some repository cleanup, in particular:
- exclude some crates from default members so they are not built and tested by default
- remove some dead code
- improve tarpaulin setup
- use new module naming convention everywhere
- consolidate diff action in CI
- update and clean up README file

## Test plan

No behavioral changes. CI changes should be tested automatically.

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -4,7 +4,6 @@ repos:
     hooks:
       - id: check-merge-conflict
       - id: check-yaml
-      - id: trailing-whitespace
```
Comment (giac-mysten) [inline .pre-commit-config.yaml:7]: Question: why remove this? covered by editorconfig?

File: .tarpaulin.toml [hunk 2]
```diff
@@ -2,12 +2,13 @@
...
   "walrus-core",
-  "walrus-event",
+  "walrus-e2e-tests",
   "walrus-sdk",
   "walrus-service",
   "walrus-sui",
+  "walrus-utils",
 ]
-run-types = ["Doctests", "Lib", "Tests"]
+run-types = ["Lib", "Tests"]
```
Comment (giac-mysten) [inline .tarpaulin.toml:11]: Questions: i guess we don't have doctests?

File: crates/walrus-service/bin/deploy.rs [hunk 3]
```diff
@@ -312,38 +311,6 @@ mod commands {
     }
 }
 
-/// Wait for SIGINT and SIGTERM (unix only).
```
Comment (giac-mysten) [inline crates/walrus-service/bin/deploy.rs:315]: question: i see this was a duplicate. Why did CI not complain?

File: .pre-commit-config.yaml (hunk 1 above)
Comment (mlegner) [reply .pre-commit-config.yaml:7]: Because we have the same a few lines below... 😅

File: .tarpaulin.toml (hunk 2 above)
Comment (mlegner) [reply .tarpaulin.toml:11]: We do, but somehow they don't seem to work in Tarpaulin atm... 🤷 

But I will have to do some additional work on this (#1139).

File: crates/walrus-service/bin/deploy.rs (hunk 3 above)
Comment (mlegner) [reply crates/walrus-service/bin/deploy.rs:315]: This *might* be due to the `#[tokio::main]` attribute, but I'm not sure. 🤷 

I believe the duplication happened when reorganizing the crate structure.

**Pull request merged**